src/Controller/Component/PaginatorComponent.php

```diff
@@ -187,9 +187,10 @@ public function paginate($object, array $settings = [])
             $query->applyOptions($options);
         }
 
+        $cleanQuery = clone $query;
         $results = $query->all();
         $numResults = count($results);
-        $count = $numResults ? $query->count() : 0;
+        $count = $numResults ? $cleanQuery->count() : 0;
```

Comment:
I'll log a ticket for 3.5/3.6 to separate the runtime and configuration state. Hopefully we can get to it then.